model/Shop/Shop.php

```diff
@@ -1073,14 +1073,20 @@ public function dejNazvyZrusenychNakupu(string $zdrojZruseni, int $rocnik = null
 
     public function prodat(int $idPredmetu, int $kusu = 1, bool $vcetneOznamemi = false)
     {
+        $cena_aktualni = dbOneCol("SELECT cena_aktualni FROM shop_predmety WHERE id_predmetu={$idPredmetu}");
+
         for ($i = 1; $i <= $kusu; $i++) {
             dbQuery(<<<SQL
 INSERT INTO shop_nakupy(id_uzivatele,id_objednatele,id_predmetu,rok,cena_nakupni,datum)
-VALUES ({$this->zakaznik->id()},{$this->objednatel->id()},{$idPredmetu},{$this->systemoveNastaveni->rocnik()},(SELECT cena_aktualni FROM shop_predmety WHERE id_predmetu={$idPredmetu}),NOW())
+VALUES ({$this->zakaznik->id()},{$this->objednatel->id()},{$idPredmetu},{$this->systemoveNastaveni->rocnik()},{$cena_aktualni},NOW())
 SQL,
             );
         }
 
+        if ($this->zakaznik->id() === 1) {
```

Comment:
Vyřešil jsi to dobře.
Použij jen prosím konstatnu `Uzivatel::SYSTEM`, aby byl záměr jasnější a snáze se hledalo, kde všude má user system dopad.